orchestrator/workflow.py

```diff
@@ -307,28 +320,29 @@ def step_group(name: str, steps: StepList, extract_form: bool = True) -> Step:
         extract_form: Whether to attach the first form of the sub steps to the step group
     """
 
+    steps = _extend_step_group_steps(name, steps)
+
     def func(initial_state: State) -> Process:
         step_log_fn = step_log_fn_var.get()
 
-        def dblogstep(step_: Step, p: Process) -> Process:
-            # Add sub_step info to state
-            p = p.map(lambda s: s | {"__sub_step": step_.name, "__step_group": name})
-            p = step_log_fn(step_, p)
-            # If this was the last sub step, remove sub_step info from state
-            step_state = p.unwrap()
-            if step_state.get("__sub_step") == steps[-1].name and step_state.get("__step_group") == name:
-                # The step group is finished. Remove sub step data from state
-                p = p.on_success(lambda s: {k: v for k, v in s.items() if k not in ["__sub_step", "__step_group"]})
-            return p
-
         # If sub_step information is present in the state. Resume from the next sub step
         if "__sub_step" in initial_state:
             step_list = StepList(dropwhile(lambda s: s.name != initial_state.get("__sub_step"), steps))[1:]
         else:
             step_list = steps
 
+        def dblogstep(step_: Step, p: Process) -> Process:
+            p = p.map(lambda s: s | {"__sub_step": step_.name, "__step_name_override": name})
```

Comment:
Can we have different names instead of reusing/overwriting p? Makes debugging easier if something goes wrong here.